internal/dockerfile/dockerfile.go

```diff
@@ -62,6 +69,17 @@ func (d Dockerfile) RmPaths(pathsToRm []string) Dockerfile {
 	return d.Join(fmt.Sprintf("RUN %s", rmCmd.String()))
 }
 
+// Pull out the `# syntax = docker/xxx` line, if one exists.
+func (d Dockerfile) MaybeSyntaxFlag() Dockerfile {
```

Comment:
this is not quite right. can you use the docker parsing code for this?
https://github.com/moby/buildkit/blob/3f8ab160d5079539f9ed971fb069e4205108dd9d/frontend/dockerfile/dockerfile2llb/directives.go#L27

Reply:
amazing, i didn't realize that existed!